File: 2. Source Code/Bmwa/Bmwa.SPA/src/app/_models/subject-program.ts
import { EducationProgram } from './education-program';
import { Subject } from './subject';
export class SubjectProgram {
    eduProgId: number;
    subjectId: number;
}

export class SucjectProgramForAdd {
    educationProgram: EducationProgram;
    subjectIDList: number[];
}

export class SucjectProgramForUpdate {
    educationProgram: EducationProgram;
    subjects: Subject[];
}

File: 2. Source Code/Bmwa/Bmwa.API/Models/SubjectProgram.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace Bmwa.API.Models
{
    [Table("SubjectProgram")]
    public class SubjectProgram
    {
        public int Id { get; set; }
        public int SubjectId { get; set; }
        public int EduProgId { get; set; }
        public Subject Subject { get; set; }
        public EducationProgram EducationProgram { get; set; }
    }
}